### DISCUSSION TITLE
Support for Javascript Frameworks

### DISCUSSION THREAD

**Discussion by binaryben** [quarto-dev/quarto-cli#4893] [Q&A] [answered]
Hey, this software looks amazing for writing academic reports. I'll definitely be using when I'm studying next.

There is [this plugin that allows Svelte components to be used](https://github.com/360-info/sverto). Could inclusion of Javascript and JS Frameworks be considered as something to add with first party support and at a more generic level? I'd love to see React, Vue and Svelte supported in particular.

It would allow two main use cases:

1. Another way of creating new modules/extensions for use in generic (HTML based) documents
    * In particular, I am hoping to create a website [like this awesome interactive maths learning website](https://mathigon.org/course/triangles/sine-cosine-rule) (for another topic) with Quarto
    * Some of the interactive components could be made in a JS Framework instead of Lua
2. Documentation for software projects
    * This already seems to have some level of support with the inclusion of Docusaurus sites as an output option

Not sure if this would be considered implemented by providing a way to use JS for Extensions or if it would be supported as a native supported language like Python and R are.

**Comment by jjallaire** [comment, marked as answer]
You should take a look at the Lua API's ability to include dependencies (described here: https://quarto.org/docs/extensions/lua-api.html#dependencies). This would allow you to e.g. enclose math in a special div that is handled specially by your JS code.

Net: you can do anything you want by including JS (w/ libraries as required) and then just applying the JS to portions of the document based on e.g. divs with special classes.

We have first-party support for Observable JS (https://quarto.org/docs/computations/ojs.html) because it aligns particularly well with scientific/technical communication. Using other frameworks is absolutely possible (as demonstrated w/ the Svelte plugin) but just requires a bit more work up front.
